Library/Homebrew/test/utils/git_spec.rb

```diff
@@ -0,0 +1,209 @@
+require "utils/git"
+
+describe Git do
+  before(:all) do
+    git = HOMEBREW_SHIMS_PATH/"scm/git"
+    file = "lib/blah.rb"
+    repo = Pathname.new("repo")
+
+    (repo/"lib").mkpath
+    system "#{git} init"
+    FileUtils.touch("repo/#{file}")
+
+    File.open(repo/file, "w") { |f| f.write("blah") }
+    system git, "add", repo/file
+    system git, "commit", "-m", "'File added'"
+    @h1 = `git rev-parse HEAD`
+
+    File.open(repo/file, "w") { |f| f.write("brew") }
```

Comment:
I'd still say this form with a previous `unlink` is probably nicer but feel free to leave as-is and just change the case above.